File: src/js/tests/TextFormattingTests.js
var textFormattingTests = function() {
	var parser;
	
	return {
		setup: function() {
			parser = createParser();
		},
		
		tearDown: function() {
			parser = null;
		},
		
		test: function Text_formatting_tests() {
			return [
				function Should_parse_strong() {
					Assert.that(trimLf(parser.parseAndReturn("__text__")), Is.equalTo("<p><strong>text</strong></p>"));
				},
				
				function Should_parse_emphasis() {
					Assert.that(trimLf(parser.parseAndReturn("''text''")), Is.equalTo("<p><em>text</em></p>"));
				},
				
				function Should_parse_underline() {
					Assert.that(trimLf(parser.parseAndReturn("--text--")), Is.equalTo("<p><ins>text</ins></p>"));
				},
				
				function Should_parse_strike_through() {
					Assert.that(trimLf(parser.parseAndReturn("---text---")), Is.equalTo("<p><del>text</del></p>"));
				},
				
				function Should_parse_teletype() {
					Assert.that(trimLf(parser.parseAndReturn("==text==")), Is.equalTo("<p><tt>text</tt></p>"));
				},
				
				function Should_parse_formatted_string_within_formatted_string() {
					Assert.that(trimLf(parser.parseAndReturn("__text ''foo''__")), Is.equalTo("<p><strong>text <em>foo</em></strong></p>"));
				}
			];
		}
	};
}();

Jarvis.run(textFormattingTests);

File: src/dotnet/Tests/ButterflyNet.Tests/TextFormattingTests.cs
using NUnit.Framework;

namespace ButterflyNet.Parser.Tests {
	[TestFixture]
	public class TextFormattingTests : WikiToHtmlTest {
		[Test]
		public void Should_parse_strong() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("__text__"), "<p><strong>text</strong></p>");
		}

		[Test]
		public void Should_parse_emphasis() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("''text''"), "<p><em>text</em></p>");
		}

		[Test]
		public void Should_parse_underline() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("--text--"), "<p><ins>text</ins></p>");
		}

		[Test]
		public void Should_parse_strikethrough() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("---text---"), "<p><del>text</del></p>");
		}

		[Test]
		public void Should_parse_teletype() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("==text=="), "<p><tt>text</tt></p>");
		}

		[Test]
		public void Should_parse_formatted_string_within_formatted_string() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("__text ''foo''__"), "<p><strong>text <em>foo</em></strong></p>");
		}

		[Test]
		public void Should_parse_big_text() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("(+text+)"), "<p><big>text</big></p>");
		}

		[Test]
		public void Should_parse_nested_big_text() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("(+text (+bigger+)+)"), "<p><big>text <big>bigger</big></big></p>");
		}

		[Test]
		public void Should_parse_small_text() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("(-text-)"), "<p><small>text</small></p>");
		}

		[Test]
		public void Should_parse_nested_small_text() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("(-text (-smaller-)-)"), "<p><small>text <small>smaller</small></small></p>");
		}

		[Test]
		public void Can_have_big_closer_without_opener() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("foo+)"), "<p>foo+)</p>");
		}

		[Test]
		public void Can_have_small_closer_without_opener() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("foo-)"), "<p>foo-)</p>");
		}

		[Test]
		public void Should_properly_parse_strikethrough_followed_by_underline() {
			AssertWithNoRegardForLineBreaks(Parser.ParseAndReturn("-----text-----"), "<p><del><ins>text</ins></del></p>");
		}
	}
}